Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+            
+            obj.updatingVal = true;
+            obj.active_module = module;
+
+            if strcmp(module, obj.moduleVIS)
+                obj.emm_setpoint = target;
+                try
+                    out = obj.com('set_wavelength', 'EMM', target, .1); % last arg is timeout
+                
+                    if out.status ~= 0
+                        error('Failed to set EMM target')
+                    end
+                catch
+                    % expected?
+                end
+            else
+                obj.emm_setpoint = NaN;
+            end
+            
+            attempting = true;
+            failcount = 0;
+            
+            while attempting
+                nir_wavelength = obj.determineNIRWavelength(target);
+                obj.solstis_setpoint = nir_wavelength;
+                out = obj.com('set_wavelength', 'solstis', nir_wavelength, 0); % last arg is timeout
+                
+                if out.status ~= 0  % Call failed
+                    failcount = failcount + 1;
+                else
+                    obj.tuning = true;
+
+                    obj.getFrequency();
+
+                    obj.trackFrequency(obj.c/target);   % Will block until obj.tuning = false (calling obj.getFrequency each tick)
+                
+                    if ~obj.center_percent || (obj.resonator_voltage < 120 && obj.resonator_voltage > 80)    % If we are good with anything or we are inside the acceptable range...
+                        attempting = false;
+                    else
+                        detuning = 1 - 2*(nir_wavelength > 900);    % Tune up if wavelength < 900, down otherwise.
+                        out = obj.com('set_wavelength', 'solstis', nir_wavelength + detuning, 0);
+                        
+                        if out.status ~= 0  % Call failed
+                            failcount = failcount + 1;
+                        end
+                        
+                        pause(.5);
+                    end
+                end
+                
+                if failcount > 10
+                    attempting = false;
+                end
+            end
+            obj.updatingVal = false;
+                
+            if failcount > 10
+                error('Failed to set the SolsTiS wavelength ten times, aborting tuning attempt.')
+            end
+                
+            if      ~obj.do_etalon_lock
+                obj.com('lock_wavelength', 'solstis', obj.lockList{1});
+                obj.com('set_etalon_lock', 'solstis', obj.lockList{1});
+            elseif  ~obj.do_wavelength_lock
+                obj.com('lock_wavelength', 'solstis', obj.lockList{1});
+            end
+            
+            obj.getFrequency();
+            
+            if strcmp(module, obj.moduleVIS) && abs(obj.VIS_wavelength - target) > .001 % If we're off with the EMM, we probably didn't initially have a good reading on the diff_wavelength.
```

Comment:
I might feel better about an arbitrary number being in frequency space, but in general, I might prefer to make any numbers like this at the very least constant class properties if not prefs

Reply:
Made constant.